src/main/java/io/strimzi/kafka/bridge/tracing/OpenTelemetryHandle.java

```diff
@@ -0,0 +1,261 @@
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import io.vertx.kafka.client.producer.KafkaHeader;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Headers;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.COMPONENT;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.JAEGER;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.KAFKA_SERVICE;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_SERVICE_NAME_PROPERTY_KEY;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_TRACES_EXPORTER_ENV_KEY;
+import static io.strimzi.kafka.bridge.tracing.TracingConstants.OPENTELEMETRY_TRACES_EXPORTER_PROPERTY_KEY;
+
+/**
+ * OpenTelemetry implementation of Tracing.
+ */
+class OpenTelemetryHandle implements TracingHandle {
+
+    static void setCommonAttributes(SpanBuilder builder, RoutingContext routingContext) {
+        builder.setAttribute("component", COMPONENT);
+        builder.setAttribute(SemanticAttributes.PEER_SERVICE, KAFKA_SERVICE);
+        builder.setAttribute(SemanticAttributes.HTTP_METHOD, routingContext.request().method().name());
+        builder.setAttribute(SemanticAttributes.HTTP_URL, routingContext.request().uri());
+    }
+
+    @Override
+    public String envName() {
+        return OPENTELEMETRY_SERVICE_NAME_ENV_KEY;
+    }
+
+    @Override
+    public String serviceName(BridgeConfig config) {
+        String serviceName = System.getenv(envName());
+        if (serviceName == null) {
+            // legacy purpose, use previous JAEGER_SERVICE_NAME as OTEL_SERVICE_NAME (if not explicitly set)
+            serviceName = System.getenv(Configuration.JAEGER_SERVICE_NAME);
+            if (serviceName == null) {
+                serviceName = config.getTracingServiceName();
+            }
+            if (serviceName != null) {
+                System.setProperty(OPENTELEMETRY_SERVICE_NAME_PROPERTY_KEY, serviceName);
+            }
+        }
+        if (serviceName != null) {
+            if (System.getenv(OPENTELEMETRY_TRACES_EXPORTER_ENV_KEY) == null && System.getProperty(OPENTELEMETRY_TRACES_EXPORTER_PROPERTY_KEY) == null) {
+                System.setProperty(OPENTELEMETRY_TRACES_EXPORTER_PROPERTY_KEY, JAEGER); // it wasn't set in script
+            }
+        }
+        return serviceName;
+    }
```

Comment:
But if the user sets `bridge.tracing=opentelemetry` in the application property and then forgets to set the `OTEL_SERVICE_NAME` env var, I don't think it makes much sense to fall back to the Jaeger service name.

Reply:
It could have that env var set from before ... and just changes tracing impl type ...
But yeah, let me know what's put final decision, and I'll adjust the code - if needed.

Reply:
> It could have that env var set from before ... and just changes tracing impl type ...

You are saying that the initial situation is:

```shell
JAEGER_SERVICE_NAME = servicename
bridge.tracing=jaeger
```

then the user just changes the `bridge.tracing=opentelemetry` but forgets to set `OTEL_SERVICE_NAME`.
Following the code, it's going to set the `otel.service.name` with the Jaeger env var.
I have tried but nothing is showed in the Jaeger UI. The service name doesn't show up in the drop down list at all.

Reply:
It should work?!
You're saying this `old school` doesn't work?

Reply:
Ok. It's working. I think that yesterday I didn't notice that the problem was on startup due to a missing dependency you have fixed in the evening.